pkg/zpdb/eviction_controller_test.go

```diff
@@ -120,6 +135,17 @@ func (c *testContext) assertDenyResponse(t *testing.T, reason string, statusCode
 	c.stop()
 }
 
+func (c *testContext) assertDenyResponseViaMarkPodAsDeleted(t *testing.T, pod string, reason string) {
+	response := c.controller.MarkPodAsDeleted(c.ctx, testNamespace, pod, "eviction-controller-test")
+	require.ErrorContains(t, response, reason)
+	c.stop()
+}
+
+func (c *testContext) assertAllowResponseViaMarkPodAsDeleted_noStop(t *testing.T, pod string) {
+	response := c.controller.MarkPodAsDeleted(t.Context(), testNamespace, pod, "eviction-controller-test")
+	require.NoError(t, response)
+}
+
```

Comment:
[nit] It's a bit unexpected to me that a method called `assertDenyResponseViaMarkPodAsDeleted` also stops the controller - I would make this explicit in the tests themselves and then the `assertAllow...` method doesn't need the `noStop` suffix.